test/plots/linePlotTests.ts

```diff
@@ -830,5 +829,121 @@ describe("Plots", () => {
+
+        assert.strictEqual(plot.downsamplingEnabled(true), plot, "enabling the downsampling option returns the plot");
+        assert.isTrue(plot.downsamplingEnabled(), "can enable the downsampling option");
+
+        plot.downsamplingEnabled(false);
+        assert.isFalse(plot.downsamplingEnabled(), "can disable the downsampling option");
+
+        svg.remove();
+      });
+
+      it("does not render points that should be removed in downsampling in horizontal line plots" , () => {
+        let data = [
+          {x: -10, y: -1}, // last element in previous bucket
+          {x: 0, y: 2}, // first element in current bucket
+          {x: 0.5, y: 1.5}, // the point to be removed 
+          {x: 1, y: 1}, // minimum y in current bucket
+          {x: 2, y: 4}, // maximum y in current bucket
+          {x: 3, y: 3}, // last elemnt in current bucket 
+          {x: 10, y: 2}, // first element in next bucket
+        ];
+        plot.addDataset(new Plottable.Dataset(data));
+
+        xScale.domain([-200, 200]);
+
+        plot.downsamplingEnabled(true);
+        plot.renderTo(svg);
+
+        let path = plot.content().select("path.line").attr("d");
+        let expectedRenderedData = [0, 1, 4, 3, 5, 6].map((d) => data[d]);
+        checkPathForDataPoints(path, expectedRenderedData);
+
+        svg.remove();
+      });
+
+      it("does not render points that should be removed in downsampling in vertical line plots", () => {
+        let data = [
+          {x: -1, y: -50}, // last element in previous bucket
+          {x: 2, y: 1}, // first element in current bucket
+          {x: 1.5, y: 1.5}, // the point to be removed
+          {x: 1, y: 2}, // minimum x in current bucket
+          {x: 4, y: 3}, // maximum x in current bucket
+          {x: 3, y: 4}, // last elemnt in current bucket
+          {x: 2, y: 100}, // first element in next bucket
+        ];
+        plot.addDataset(new Plottable.Dataset(data));
+
+        yScale.domain([-200, 200]);
+
+        plot.downsamplingEnabled(true);
+        plot.renderTo(svg);
+
+        let path = plot.content().select("path.line").attr("d");
+        let expectedRenderedData = [0, 1, 3, 4, 5, 6].map((d) => data[d]);
+        checkPathForDataPoints(path, expectedRenderedData);
+
+        svg.remove();
+      });
+
+      it("does not render points that are on the same line except for the first, the last, the largest and the smallest points", () => {
+        let data = [
+          {x: 3, y: 1}, // last element in previous bucket
+          {x: 2, y: 2}, // first element in the bucket
+          {x: 1, y: 1}, // minimum element in the bucket
+          {x: 10, y: 10}, // maximum element in the bucket
+          {x: 2.5, y: 2.5}, // the point to be removed
+          {x: 3, y: 3}, // last element in the bucket
+          {x: 3, y: 1}, // first element in next bucket
+        ];
+        plot.addDataset(new Plottable.Dataset(data));
+
+        yScale.domain([0, 10]);
```

Comment:
my question is that why it doesn't work with auto domain? why does it have to be [0, 10]?